File: src/traffic/impl/NetworkTrafficMonitor.php
<?php

namespace pocketcloud\cloud\traffic\impl;

use Closure;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\traffic\TrafficMonitor;
use pocketcloud\cloud\traffic\TrafficMonitorManager;
use pocketcloud\cloud\util\net\Address;
use pocketcloud\cloud\util\PathUtils;

final class NetworkTrafficMonitor extends TrafficMonitor {

    public const string NETWORK_MODE_PACKET_IN = "packet_in";
    public const string NETWORK_MODE_PACKET_OUT = "packet_out";

    public function __construct() {
        parent::__construct(TrafficMonitorManager::TRAFFIC_NETWORK);
    }

    /**
     * @param string $packetClass
     * @param Closure(CloudPacket $packet, Address $address): void $handler
     * @return self
     */
    public function monitorPacketIn(string $packetClass, Closure $handler): self {
        $this->addHandler(self::parsePacketMode(self::NETWORK_MODE_PACKET_IN, $packetClass), $handler);
        return $this;
    }

    /**
     * @param string $packetClass
     * @param Closure(CloudPacket $packet, Address $destination, bool $success): void $handler
     * @return self
     */
    public function monitorPacketOut(string $packetClass, Closure $handler): self {
        $this->addHandler(self::parsePacketMode(self::NETWORK_MODE_PACKET_OUT, $packetClass), $handler);
        return $this;
    }

    public static function parsePacketMode(string $normalMode, string $packetClass): string {
        return $normalMode . "-" . basename(PathUtils::normalize($packetClass));
    }
}

File: src/main/java/de/pocketcloud/cloud/traffic/impl/NetworkTrafficMonitor.java
package de.pocketcloud.cloud.traffic.impl;

import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.Packet;
import de.pocketcloud.cloud.traffic.TrafficMonitor;
import de.pocketcloud.cloud.traffic.TrafficMonitorManager;
import de.pocketcloud.cloud.util.TriConsumer;

import java.net.SocketAddress;

public final class NetworkTrafficMonitor extends TrafficMonitor {

    public static final String NETWORK_MODE_PACKET_IN = "packet_in";
    public static final String NETWORK_MODE_PACKET_OUT = "packet_out";

    public NetworkTrafficMonitor() {
        super(TrafficMonitorManager.TRAFFIC_NETWORK);
    }

    public <T extends CloudPacket> NetworkTrafficMonitor monitorPacketIn(Class<T> packetClass, TriConsumer<SocketAddress, T, Long> handler) {
        addHandler(parsePacketMode(NETWORK_MODE_PACKET_IN, packetClass), (a, b, by) -> {
            @SuppressWarnings("unchecked")
            T buffer = (T) b;
            handler.accept(a, buffer, by);
        });
        return this;
    }

    public <T extends CloudPacket> NetworkTrafficMonitor monitorPacketOut(Class<T> packetClass, TriConsumer<SocketAddress, T, Long> handler) {
        addHandler(parsePacketMode(NETWORK_MODE_PACKET_OUT, packetClass), (a, b, by) -> {
            @SuppressWarnings("unchecked")
            T buffer = (T) b;
            handler.accept(a, buffer, by);
        });
        return this;
    }

    public static String parsePacketMode(String normalMode, Class<? extends Packet> packetClass) {
        return normalMode + "-" + packetClass.getSimpleName();
    }
}